ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/state/BeaconState.java

```diff
@@ -193,6 +193,8 @@ static BeaconState create(
             });
   }
 
+  TransitionCaches getTransitionCaches();
```

Comment:
Probably shouldn't add this here.  `BeaconStateCache.getTransitionCaches` is used to get the cache everywhere else.

I'm not against changing this but if we do we should change it everywhere and I seem to recall @Nashatyrev had a preference for this approach and some reasoning behind it.

Reply:
I'm pretty fine with adding `BeaconState.getTransitionCaches()`, though you would either remove `BeaconStateCache` then or do ` BeaconState extends  BeaconStateCache` and yes then you would need to refactor all the usages

Reply:
Turns out `BeaconStateCache.getTransitionCaches()` works perfectly for my use case. I've removed this line.